conans/client/graph/proxy.py

```diff
@@ -142,3 +142,16 @@ def _download(self, ref, remote):
         output = ConanOutput(scope=str(ref))
         output.info("Downloaded recipe revision %s" % ref.revision)
         return recipe_layout
+
+
+def should_update_reference(reference, update):
+    if update is None:
+        return False
+    if not isinstance(update, list):
+        assert False
+    # Legacy syntax had --update without pattern, that manifests as a True in the list
+    # (Usually one element, but support having more than one just in case)
+    # Note that it can't be false, the True is a flag set by Conan to update all
+    if any(isinstance(u, bool) for u in update):
+        return True
```

Comment:
This still reads weird imho, iterating a list for a bool existence irrespective of the content is at least confusing.

I think this can be solved in origin, converting ``[True]`` from the ``const=True`` to just ``True`` to pass to the parameter
So ``update`` can be boolean or list. Otherwise, lets make it always a list, but convert in origin ``[True]`` to ``["*"]`` or something like that?

Reply:
Sounds like a good idea, will do